core/client/app/components/gh-tags-input.js

```diff
@@ -0,0 +1,270 @@
+/* global Bloodhound, key */
+import Ember from 'ember';
+
+/**
+ * Ghost Tag Input Component
+ *
+ * Creates an input field that is used to input tags for a post.
+ * @param  {Boolean} hasFocus       Whether or not the input is focused
+ * @param  {DS.Model} post          The current post object to input tags for
+ */
+export default Ember.Component.extend({
+    classNames: ['gh-input'],
+    classNameBindings: ['hasFocus:focus'],
+
+    // Uses the Ember-Data store directly, as it needs to create and get tag records
+    store: Ember.inject.service(),
+
+    hasFocus: false,
+    post: null,
+    highlightIndex: null,
+
+    isDirty: false,
+    isReloading: false,
+
+    unassignedTags: Ember.A(), // tags that AREN'T assigned to this post
+    currentTags: Ember.A(), // tags that ARE assigned to this post
+
+    // Input field events
+    click: function () {
+        this.$('#tag-input').focus();
+    },
+
+    focusIn: function () {
+        this.set('hasFocus', true);
+        key.setScope('tags');
+    },
+
+    focusOut: function () {
+        this.set('hasFocus', false);
+        key.setScope('default');
+        this.set('highlightIndex', null);
+        // if there is text in the input field, create a tag with it
+        if (this.$('#tag-input').val() !== '') {
+            this.send('addTag', this.$('#tag-input').val());
+        }
+        this.saveTags();
+    },
+
+    // Tag Loading functions
+    loadTagsOnInit: Ember.on('init', function () {
+        var self = this;
+
+        if (this.get('post')) {
+            this.loadTags().then(function () {
+                Ember.run.schedule('afterRender', self, 'initTypeahead');
+            });
+        }
+    }),
+
+    reloadTags: Ember.observer('post', function () {
+        var self = this;
+
+        this.loadTags().then(function () {
+            self.reloadTypeahead(false);
+        });
+    }),
+
+    loadTags: function () {
+        var self = this,
+            post = this.get('post');
+
+        this.get('currentTags').clear();
+        this.get('unassignedTags').clear();
+
+        return this.get('store').find('tag', {limit: 'all'}).then(function (tags) {
+            if (post.get('id')) { // if it's a new post, it won't have an id
+                self.get('currentTags').pushObjects(post.get('tags').toArray());
+            }
+
+            tags.forEach(function (tag) {
+                if (Ember.isEmpty(post.get('id')) || Ember.isEmpty(self.get('currentTags').findBy('id', tag.get('id')))) {
+                    self.get('unassignedTags').pushObject(tag);
+                }
+            });
+
+            return Ember.RSVP.resolve();
+        });
+    },
+
+    // Key Binding functions
+    bindKeys: function () {
+        var self = this;
+
+        key('enter, tab, ,', 'tags', function (event) {
```

Comment:
Does this take into account that the comma is different in different locales? The original code was:

`isComma = ','.localeCompare(String.fromCharCode(event.keyCode || event.charCode)) === 0;`

This has regressed twice before, so I'm keen to avoid it happening again.